services/transaction/shared/transaction.js

```diff
@@ -0,0 +1,56 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2021 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const { v4: uuidv4 } = require('uuid');
+const {
+	Exceptions: { ServiceUnavailableException },
+} = require('lisk-service-framework');
+
+const mysqlIndex = require('./indexdb/mysql');
+const MultisignatureTxIndexSchema = require('./schema/multisignature');
+
+const getMultiSignatureTxIndex = () => mysqlIndex('MultisignatureTx', MultisignatureTxIndexSchema);
+
+const createMultisignatureTx = async inputTransaction => {
+	const multisignatureTxDB = await getMultiSignatureTxIndex();
+	const transaction = {
+		data: [],
+		meta: {},
+	};
+
+	// TODO: Place holder for actual implementation
+	// Return mock response for now
+	inputTransaction = {
+		nonce: 1,
+		senderPublicKey: '3e50549cd4d98760064ff2fe51801afba4e5e8623335275cece0eeff8495a81b',
+		asset: '{"numberOfSignatures":2,"mandatoryKeys":["228c865b903dab827342aa6611676bf883e982e7cd467c9168a7966cdabb391c","9bc945f92141d5e11e97274c275d127dc7656dda5c8fcbf1df7d44827a732664"],"optionalKeys":[]}',
+		moduleAssetId: '4:0',
+		fee: '314000',
+		rejected: false,
+	};
+
+	transaction.data.push({ ...inputTransaction, serviceId: uuidv4() });
```

Comment:
`serviceId` needs to be computed deterministically based on the following statement from the [Lisk Service Multisignature Pooling Design Proposal](https://docs.google.com/document/d/1DJV0D2cWZGCOzyGr-i09SFpimNXo3B-bgC6H8DKvo5w/edit#heading=h.7e91hnkuvxpa):
```
The transaction ID can be generated by Lisk Elements based on the following properties:
nonce, senderPublicKey, moduleAssetId, fee, asset(amount, recipientId).
```

The current approach to computing the `serviceId` is indeterministic. Kindly consider generating a (deterministic) UUID based on a seed string that is formed from the required properties. We can discuss more on this offline, if necessary.

Reply:
I had an offline discussion with @MichalTuleja regarding this and we agreed on generating any random `UUID` instead of generating based on transaction properties.
What do you say @MichalTuleja ?

Reply:
It seems we've got a discrepancy between the specs and it wasn't properly addressed in the GH issue. Thanks for reminding about it @sameersubudhi.

The usage of a random UUID sounds like a reasonable solution to me, however assuming we agreed to make it deterministic leaves us no other choice.

@nagdahimanshu I would suggest the following: we close that issue with the current implementation and continue work on the custom UUID generator in another issue.